prow/git/v2/interactor.go

```diff
@@ -278,6 +282,32 @@ func (i *interactor) FetchRef(refspec string) error {
 	return nil
 }
 
+// FetchFromRemote fetches all update from a specific remote and branch and leaves it as FETCH_HEAD.
+func (i *interactor) FetchFromRemote(remote RemoteResolver, branch string) error {
+	r, err := remote()
```

Comment:
We only read this value once and don't store it -- why not just pass the URI as a string? What benefit is it to pass a resolver?

Reply:
Yea, we could do that. Since I already here parse the url.